owasp-top10-2021-apps/a9/games-irados/app/routes.py

```diff
@@ -56,6 +57,7 @@ def login_required(f):
     def decorated_function(*args, **kwargs):
         if 'username' not in session:
             flash('oops, session expired', "danger")
+            app.logger.info('Session expired.')
```

Comment:
Nice, but for what is the user IP address? What about the logout function?

Reply:
Thanks! fixed.